pkg/datapath/config/gen.go

```diff
@@ -13,3 +13,4 @@ package config
 //go:generate go run github.com/cilium/cilium/tools/dpgen -path ../../../bpf/bpf_overlay.o -embed Node -kind object -name BPFOverlay -out overlay_config.go
 //go:generate go run github.com/cilium/cilium/tools/dpgen -path ../../../bpf/bpf_network.o -embed Node -kind object -name BPFNetwork -out network_config.go
 //go:generate go run github.com/cilium/cilium/tools/dpgen -path ../../../bpf/bpf_wireguard.o -embed Node -kind object -name BPFWireguard -out wireguard_config.go
+//go:generate go run github.com/cilium/cilium/tools/dpgen -path ../../../bpf/bpf_sock.o -kind object -name BPFSocket -out sock_config.go
```

Comment:
Lets keep the `Node` object embedded as with the other configuration structs (mainly for future use) and keep the naming in line with the C/.o bpf files:
```suggestion
//go:generate go run github.com/cilium/cilium/tools/dpgen -path ../../../bpf/bpf_sock.o -embed Node -kind object -name BPFSock -out sock_config.go
```

Reply:
understood

Reply:
done, but had to export/move `nodeConfig()` function to avoid import cycle loop :/